cmd/ova-provider-server/ova-provider-server.go

```diff
@@ -466,8 +467,27 @@ func convertToVmStruct(envelope []Envelope, ovaPath []string) ([]VM, error) {
 					newVM.MemoryUnits = item.AllocationUnits
 
 				} else {
+					var itemKind string
+					if len(item.ElementName) == 0 {
+						if len(item.Description) > 0 {
+							itemKind = item.Description
+						} else {
+							itemKind = "Unknown"
+						}
+					} else {
+						// if the `ElementName` element has a name such as "Hard Disk 1", strip off the
+						// number suffix
+						runes := []rune(item.ElementName)
+						if len(runes) > 2 &&
+							unicode.IsSpace(runes[len(runes)-2]) &&
+							unicode.IsDigit(runes[len(runes)-1]) {
+							itemKind = item.ElementName[:len(item.ElementName)-2]
+						} else {
+							itemKind = item.ElementName
+						}
```

Comment:
```suggestion
itemKind = strings.TrimRightFunc(item.ElementName, func(r rune) bool {
        return unicode.IsDigit(r) || unicode.IsSpace(r)
    })
```

IMHO using trim right is nicer here:
 it removes all trailing spaces and digits, not just a specific pattern.
 it is more concise and uses standard library functions, making it easier to read and understand.
 it does not require custom logic
 
using `strings.TrimRightFunc` might be slower (?) but IMHO we can ignore this penalty in our use case.

Reply:
p.s. in both implementations we need to check that the resulting `itemKind` after removing the number suffix is not empty.